Sphinx config and updated docs for device plugin

AMD-melliott opened the pull request:
This pull request contains:
- Updated sphinx config for [instinct.docs.amd.com](https://instinct.docs.amd.com)
- Updated device plugin docs and example files to match tone and formatting of the [GPU Operator docs](https://instinct.docs.amd.com/projects/gpu-operator/en/latest/)

This PR is related to internal Jira DCGPUBU-561

AMD-melliott commented:
Leaving this in draft state since there are a few things I'd like to validate. Please provide feedback and input on anything else that needs to be added.

y2kenny-amd reviewed (commented):
Thanks for adding the much needed documentations.  I left a couple of comments.

AMD-melliott commented:
@y2kenny-amd Thanks for the comments. I've added some additional edits/info and updated some of the provided examples. I've tested all of this on the TME server, but would appreciate review and feedback.

@farshadghodsian heads up that I've moved this PR out of draft state

AMD-melliott commented on README.md:
```diff
@@ -1,11 +1,13 @@
 
 This is a [Kubernetes][k8s] [device plugin][dp] implementation that enables the registration of AMD GPU in a container cluster for compute workload.  With the appropriate hardware and this plugin deployed in your Kubernetes cluster, you will be able to run jobs that require AMD GPU.
 
+This plugin is required by tools such as the [AMD GPU Operator](https://github.com/ROCm/gpu-operator) to expose AMD GPUs as schedulable resources.
```
I'm working on a lot of getting started content, so I'd like to leave this information in. As the story gets more complicated, I want to create a holistic overview of how all of the Kubernetes components interact. Having this here is a breadcrumb to help new users understand how the pieces fit together.

yansun1996 commented on docs/contributing/development.md:
```diff
@@ -0,0 +1,52 @@
+
+Before you begin development, ensure you have the following prerequisites:
+
+- A working installation of Go (version 1.16 or later).
```
I think the current device plugin repo has claimed go 1.23 as the current golang version. what do you think @AMD-melliott @y2kenny-amd ?

y2kenny replied:
This is correct.  go.mod points to 1.23.

yansun1996 commented on docs/index.md:
```diff
@@ -1,4 +1,80 @@
+
+## Example Workload
+
+You can restrict workloads to a node with a GPU by adding `resources.limits` to the pod definition. An example pod definition is provided in [example/pod/pytorch.yam](https://raw.githubusercontent.com/ROCm/k8s-device-plugin/master/example/pod/pytorch.yaml)l. Create the pod by running:
```
```example/pod/pytorch.yam``` ---> ```example/pod/pytorch.yaml```

farshadghodsian commented on docs/user-guide/installation.md:
````diff
@@ -0,0 +1,213 @@
+kubectl create -f https://raw.githubusercontent.com/ROCm/k8s-device-plugin/master/k8s-ds-amdgpu-dp.yaml
+```
+
+### Option 2: Device Plugin with Health Checks
````
Can we also include a 3rd option to install Device Plugin with GPU Operator and then link to the GPU Operator site?
